subprojects/language-native/src/main/java/org/gradle/language/swift/plugins/SwiftBasePlugin.java

```diff
@@ -76,5 +90,25 @@ public void apply(ProjectInternal project) {
         nativeRuntime.extendsFrom(implementation);
         nativeRuntime.setCanBeConsumed(false);
         nativeRuntime.getAttributes().attribute(Usage.USAGE_ATTRIBUTE, project.getObjects().named(Usage.class, Usage.NATIVE_RUNTIME));
+
+        Configuration testImplementation = project.getConfigurations().create(TEST_IMPLEMENTATION);
+        testImplementation.extendsFrom(implementation);
```

Comment:
This should be a _dependency_ on the main component. Can you add a TODO for us to fix this later?

We'll also need some test coverage to show that this works, for example, show that the unit tests can have additional dependencies that the main component does not. Can add this in a later PR